content/ADR-201-lambdas-wearables-API-analysis.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: adr
+adr: 201
+title: Lambdas Wearables API Analysis
+date: 2020-04-03
+status: Draft
+type: RFC
+spdx-license: CC0-1.0
+authors:
+  - pedrotambo
+---
+
+## Abstract
+
+The Explorer uses the Lambdas API `/collections/wearables` to get Wearables. This API sometimes has bad performance, it has an ill design and the handrail between the Explorer and the Content-Server through Lambdas might not be necessary. In this document we review the functionalities of this API and analize solutions to this problems. 
+
+## Context, Reach & Prioritization
+
+During the last MVFW, a report of bad performance APIs was done. Among those APIs, there was the Lambdas API `GET /collections/wearables`.
+
+### API Funcionality
+First, a review of the functionalities of the API will be covered. The API, based on the request filters, gets the wearables from the content server, applies a small transform and returns the result to the client.
+
+#### Request filters
+The API allows three query parameters:
+
+- `wearableIds`: A list of wearable URNs that the client requests. Just ask for them to the content server and apply the transform. Sometimes many wearables are requested that the url string exceeds the maximum allowed.
+- `collectionIds`: A list of collection URNs that the client requests. For a `collectionId` is "off-chain" (base-avatars only), request the wearables directly to the content-server. If a *collectionId* is "on-chain" (collections v1 and v2), request to The Graph the wearable URNs from that collection, and then request them to the content-server. No third-party collections are included. Note: for the base-avatars collection it has a fixed list in a file listing all the urns.
+- `textSearch`: A string value that is used to query The Graph. On-chain wearables have a metadata field called "searchText" that can have anything that is used for filtering. In some cases it has useful information but in other cases it has non sense info.
+
+#### Transform: Wearables from the content-server to the Client
+
+1. First the wearable data from the content-server is retrieved with this [schema](https://github.com/decentraland/schemas/blob/main/src/platform/entity.ts#L28-L37).
+2. Then only the `metadata` of type [Wearable](https://github.com/decentraland/schemas/blob/main/src/platform/item/wearable/wearable.ts#L18-L26) is returned to the client, but a small transform is applied: the contents are extended with full url.
+
+![Entity](/resources/ADR-201/entity-transform.png)
```

Comment:
I think this is not rendering, should it be?:
```suggestion
![Entity](/public/resources/ADR-201/entity-transform.png)
```

Reply:
It is, the rendering process fixes the url correctly.